homeassistant/components/ezviz/config_flow.py

```diff
@@ -0,0 +1,261 @@
+"""Config flow for ezviz."""
+import logging
+
+from pyezviz import EzvizClient
+from pyezviz.client import PyEzvizError
+from requests.exceptions import HTTPError
+import voluptuous as vol
+
+from homeassistant import exceptions
+from homeassistant.config_entries import CONN_CLASS_CLOUD_POLL, ConfigFlow, OptionsFlow
+from homeassistant.const import (
+    CONF_PASSWORD,
+    CONF_REGION,
+    CONF_TIMEOUT,
+    CONF_TYPE,
+    CONF_USERNAME,
+)
+from homeassistant.core import callback
+from homeassistant.data_entry_flow import AbortFlow
+
+from .const import (  # pylint: disable=unused-import
+    ATTR_SERIAL,
+    ATTR_TYPE_CAMERA,
+    ATTR_TYPE_CLOUD,
+    CONF_FFMPEG_ARGUMENTS,
+    DEFAULT_CAMERA_USERNAME,
+    DEFAULT_FFMPEG_ARGUMENTS,
+    DEFAULT_REGION,
+    DEFAULT_TIMEOUT,
+    DOMAIN,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class EzvizConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Ezviz."""
+
+    VERSION = 1
+    CONNECTION_CLASS = CONN_CLASS_CLOUD_POLL
+
+    async def _validate_and_create_auth(self, data):
+        """Validate the user input allows us to connect.
+
+        Data has the keys from data_schema with values provided by the user.
+        """
+        await self.async_set_unique_id(data[CONF_USERNAME])
+        self._abort_if_unique_id_configured()
+
+        # constructor does login call
```

Comment:
Is this comment still true?

Reply:
Still true for the Ezviz HA domain. There can be only one Ezviz cloud account linked to a specific email address.

Could this cause issues if a user has a cloud account, in a different domain, linked to their email address? (Blink or canary for example. - if they where also using email address as unique id)

Reply:
If creating the instance makes I/O we need to schedule the creation on the executor. But it doesn't look like there are any side effects in the init method. 

https://github.com/BaQs/pyEzviz/blob/39e2b0cbf85d246cdd5510e24b64eda5ea9ed48f/pyezviz/client.py#L37-L54

Reply:
I'm changing self._session via methods but this isn't really IO, it just sets a cookie. (This is a side effect if I understand the term correctly)

Reply:
I'm not following. I'm asking if there are any side effects when we instantiate the client class. I can't see any. Please point to where that is if there are side effects at this point.

Reply:
There isn't any when initiating the class. Apologies for the misunderstanding.

Reply:
Good. Then we can remove this comment.